mlflow/utils/environment.py

```diff
@@ -80,5 +85,40 @@ def _get_additional_pip_dep(conda_env):
 
 
 def _log_pip_requirements(conda_env, path, requirements_file="requirements.txt"):
-    pip_deps = _get_additional_pip_dep(conda_env)
+    pip_deps = _get_pip_dep(conda_env)
     _mlflow_additional_pip_env(pip_deps, path=os.path.join(path, requirements_file))
+
+
+def _parse_pip_requirements(pip_requirements):
+    if isinstance(pip_requirements, str):
+        return [
+            req.requirement for req in parse_requirements(pip_requirements, session=PipSession())
+        ]
+    elif isinstance(pip_requirements, list):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            tmp_path = os.path.join(tmpdir, f"{uuid.uuid4().hex}.txt")
+            with open(tmp_path, "w") as f:
+                f.write("\n".join(pip_requirements))
+            return _parse_pip_requirements(tmp_path)
```

Comment:
The current implementation fails in the following case

```python
def test_parse_pip_requirements_chdir(request, tmpdir):
    try:
        os.chdir(tmpdir)
        f = tmpdir.join("requirements.txt")
        f.write("b")
        assert "requirements.txt" in os.listdir()
        assert _parse_pip_requirements(f.strpath) == ["b"]
        assert _parse_pip_requirements(["a", "-r requirements.txt"]) == ["a", "b"]
    finally:
        os.chdir(request.config.invocation_dir)
```

```
url = '/var/folders/33/kkm875tn7cxb89b8mmcykrx00000gn/T/tmp68onj37q/requirements.txt'
session = <pip._internal.network.session.PipSession object at 0x7fd0b06d1650>

    def get_file_content(url, session):
        # type: (str, PipSession) -> Tuple[str, str]
        """Gets the content of a file; it may be a filename, file: URL, or
        http: URL.  Returns (location, content).  Content is unicode.
        Respects # -*- coding: declarations on the retrieved files.
    
        :param url:         File path or url.
        :param session:     PipSession instance.
        """
        scheme = get_url_scheme(url)
    
        if scheme in ['http', 'https']:
            # FIXME: catch some errors
            resp = session.get(url)
            raise_for_status(resp)
            return resp.url, resp.text
    
        elif scheme == 'file':
            url = url_to_path(url)
    
        try:
            with open(url, 'rb') as f:
                content = auto_decode(f.read())
        except OSError as exc:
            raise InstallationError(
>               f'Could not open requirements file: {exc}'
            )
E           pip._internal.exceptions.InstallationError: Could not open requirements file: [Errno 2] No such file or directory: '/var/folders/33/kkm875tn7cxb89b8mmcykrx00000gn/T/tmp68onj37q/requirements.txt'
```